CMakeLists.txt

```diff
@@ -246,7 +246,10 @@ endif()
 if (S2_USE_SYSTEM_INCLUDES)
     target_include_directories(s2 SYSTEM PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/src)
```

Comment:
Should this be the same as the other branch just with `SYSTEM`?

Reply:
Do you mean should line 247 be changed to this:

```
    target_include_directories(s2 SYSTEM PUBLIC
      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
      $<INSTALL_INTERFACE:include>
    )
```

Then I would suggest the answer is "maybe, but probably no".  The addition of `S2_USE_SYSTEM_INCLUDES` probably means that someone is embedding this project inside their own project, perhaps as a git submodule, and then incorporating it into their project as a whole, meaning their resulting project builds s2 as well as their project. However, because the s2 includes are inside their own source tree, CMake will by default treat them as non-system include folders and generate warnings both during the build of s2 AND during the build of whatever downstream project includes the s2 headers.  If the downstream project has a policy of allowing zero warnings then they'd need this option to suppress them.  

The upshot is that anyone who is using this codepath is unlikely to be using the install target at all and I don't want to try to fix something that isn't currently broken.